src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/SphKerrSchild.cpp

```diff
@@ -0,0 +1,1053 @@
+    const gsl::not_null<CachedBuffer*> cache,
+    internal_tags::l_upper<DataType, Frame> /*meta*/) const {
+  const auto& kerr_schild_l =
+      cache->get_var(*this, internal_tags::kerr_schild_l<DataType, Frame>{});
+  const auto& inv_jacobian =
+      cache->get_var(*this, internal_tags::inv_jacobian<DataType, Frame>{});
+
+  l_upper->get(0) = -1.;  // this is l^t
+
+  for (size_t j = 0; j < 3; ++j) {
+    l_upper->get(j + 1) = inv_jacobian.get(0, j) * kerr_schild_l.get(0);
+    for (size_t i = 1; i < 3; ++i) {
+      l_upper->get(j + 1) += inv_jacobian.get(i, j) * kerr_schild_l.get(i);
+    }
+  }
+}
+
+template <typename DataType, typename Frame>
+void SphKerrSchild::IntermediateComputer<DataType, Frame>::operator()(
+    const gsl::not_null<tnsr::I<DataType, 3, Frame>*> deriv_r,
+    const gsl::not_null<CachedBuffer*> cache,
+    internal_tags::deriv_r<DataType, Frame> /*meta*/) const {
+  const auto& r = get(cache->get_var(*this, internal_tags::r<DataType>{}));
+  const auto& a_dot_x =
+      get(cache->get_var(*this, internal_tags::a_dot_x<DataType>{}));
+  const auto& kerr_schild_x =
+      cache->get_var(*this, internal_tags::kerr_schild_x<DataType, Frame>{});
+  const auto& H = cache->get_var(*this, internal_tags::H<DataType>{});
+  const auto spin_a = solution_.dimensionless_spin() * solution_.mass();
+
+  // temp_mass used to convert mass from type double to Scalar
+  const auto temp_mass = make_with_value<Scalar<DataType>>(
+      get_size(get_element(kerr_schild_x, 0)), solution_.mass());
+  const auto deriv_r_denom = get(H) / get(temp_mass);
+
+  for (size_t i = 0; i < 3; ++i) {
+    deriv_r->get(i) =
+        deriv_r_denom *
+        (kerr_schild_x.get(i) + a_dot_x * gsl::at(spin_a, i) / square(r));
+  }
+}
+
+template <typename DataType, typename Frame>
+void SphKerrSchild::IntermediateComputer<DataType, Frame>::operator()(
+    const gsl::not_null<tnsr::I<DataType, 4, Frame>*> deriv_H,
+    const gsl::not_null<CachedBuffer*> cache,
+    internal_tags::deriv_H<DataType, Frame> /*meta*/) const {
+  const auto& r = get(cache->get_var(*this, internal_tags::r<DataType>{}));
+  const auto& a_dot_x =
+      get(cache->get_var(*this, internal_tags::a_dot_x<DataType>{}));
+  const auto& H = cache->get_var(*this, internal_tags::H<DataType>{});
+  const auto& jacobian =
+      cache->get_var(*this, internal_tags::jacobian<DataType, Frame>{});
+  const auto spin_a = solution_.dimensionless_spin() * solution_.mass();
+  const auto& deriv_r =
+      cache->get_var(*this, internal_tags::deriv_r<DataType, Frame>{});
+
+  const auto H_denom = 1. / (pow(r, 4) + square(a_dot_x));
+  const auto factor = 3. / r - 4. * cube(r) * H_denom;
+
+  deriv_H->get(0) = 0.;  // set time component to 0
+  for (size_t i = 0; i < 3; ++i) {
+    deriv_H->get(i + 1) =
+        get(H) *
+        (factor * deriv_r.get(i) - 2. * H_denom * a_dot_x * gsl::at(spin_a, i));
+  }
+
+  const auto deriv_H_x = deriv_H->get(1);
```

Comment:
`const auto&`

Reply:
Not needed because components modified in the loop below